### PR TITLE
Upgrade yard

### PR DESCRIPTION
**Pull request by xaviershay**
Motivated by github telling me the current version has a CVE :(

Does include some minor changes to formatting/metadata (motivated by warnings) that I think is more correct than it was.

### CODE REVIEW

**Comment by xaviershay** [comment]
Ready for review, added two more commits since yesterday.

Merge on green if no suggested edits.

File: lib/rspec/core/configuration.rb [hunk 1]
```diff
@@ -1531,12 +1528,12 @@ def self.delegate_to_ordering_manager(*methods)
...
       end
 
-      # @macro delegate_to_ordering_manager
+      # @!method seed=(value)
       #
       # Sets the seed value and sets the default global ordering to random.
       delegate_to_ordering_manager :seed=
 
-      # @macro delegate_to_ordering_manager
+      # @!method seed
```
Comment (myronmarston) [inline lib/rspec/core/configuration.rb]: This is doc'ing as a `void` return type.  Do we need a `@return` to doc it properly?

**Review by myronmarston — APPROVED** [review]
Thanks @xaviershay!  It's great to get this dependency upgraded and to address some doc issues.

I left one comment but don't consider it a merge blocker since it's already an issue before your changes.

Are you planning to upgrade `yard` in the other repos?  It would be nice to get them all on the same version.  Also, we should see if our `rspec-dev` doc tooling (e.g. the `rake update_docs[version,branch,website_path]` task) works with this version of yard.  I don't think that needs to block merge, but it'll need to be addressed by somebody before we release 3.8.

File: lib/rspec/core/configuration.rb (hunk 1 above)
Comment (xaviershay) [reply lib/rspec/core/configuration.rb]: yes. I'm going to leave that out of this PR for now - there are many methods like this that are doc'ed as `void`.

**Comment by xaviershay** [comment]
Yep, will upgrade the others. Wanted to learn what it was going to look like by doing this one first :) We're also on an old version of rubocop I'd like to upgrade.

Will check the `rspec-dev` tasks you mentioned.

**Comment by xaviershay** [comment]
confirmed that `rspec-dev` still appears to work.

**Comment by myronmarston** [comment]
> confirmed that rspec-dev still appears to work.

We should also check what the generated docs look like in the context of running the rspec site locally. @JonRowe  had to hack the js/css a bit to make the docs "mount" at our desired path for our site properly and given that the new docs look different it might have broken those.  @JonRowe do you have time to take a look?

File: lib/rspec/core/configuration.rb (hunk 1 above)
Comment (xaviershay) [reply lib/rspec/core/configuration.rb]: https://github.com/rspec/rspec-core/issues/2494 for follow on

**Comment by xaviershay** [comment]
> given that the new docs look different

... AFAICT `yard` is basically generating the same HTML/styling. Running locally appears to look sane and correctly styled:

screen shot 2017-12-31 at 12 55 42 pm

**Comment by xaviershay** [comment]
Screenshot from local server, looks sane to me:

screen shot 2017-12-31 at 12 58 48 pm

**Comment by myronmarston** [comment]
Thanks for checking on that!  It seemed to look different when I was running a local yard server for some reason, but I guess that's different from how it looks rendered on our site.

**Comment by xaviershay** [comment]
Yeah I think that one is fine too:
screen shot 2017-12-31 at 6 20 51 pm

**Comment by JonRowe** [comment]
Looks fine from the diff, the "hack" is for the js/css when publishing to our rspec.info site IIRC

**Pull request merged**